azext_iot/central/_help.py

```diff
@@ -44,6 +57,288 @@ def load_central_help():
+            }"
+
+        - name: Creat a blob stoarge export destination with json payload
+          text: >
+            az iot central export destination create
+            --app-id {appid}
+            --dest-id {destintionid}
+            --content "{
+              'displayName': 'Blob Storage',
+              'type': 'blobstorage@v1',
+              'authorization': {
+                'type': 'connectionString',
+                'connectionString': 'DefaultEndpointsProtocol=https;AccountName=[accountName];AccountKey=[key];EndpointSuffix=core.windows.net',
+                'container': 'test'
+              }
+            }"
+
+        - name: create a Azure Data Explorer export destination with json payload
+          text: >
+            az iot central export destination create
+            --app-id {appid}
+            --dest-id {destintionid}
+            --content "{
+              'displayName': 'Azure Data Explorer',
+              'type': 'dataexplorer@v1',
+              'clusterUrl': 'https://[clusterName].westus2.kusto.windows.net',
+              'database': 'database1',
+              'table': 'table1',
+              'authorization': {
+                'type': 'servicePrincipal',
+                'clientId': '3b420743-2020-44c6-9b70-cc42f945db0x',
+                'tenantId': '72f988bf-86f1-41af-91ab-2d7cd011db47',
+                'clientSecret': '[Secret]'
+              }
+            }"
+
+        - name: create an Event Hub export destination with json payload
+          text: >
+            az iot central export destination create
+            --app-id {appid}
+            --dest-id {destintionid}
+            --content "{
+              'displayName': 'Event Hub',
+              'type': 'eventhubs@v1',
+              'authorization': {
+                'type': 'connectionString',
+                'connectionString': 'Endpoint=sb://[hubName].servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=*****;EntityPath=entityPath1'
+              }
+            }"
+
+        - name: create an Service Bus Queue destination with json payload
+          text: >
+            az iot central export destination create
+            --app-id {appid}
+            --dest-id {destintionid}
+            --content "{
+              'displayName': 'Service Bus Queue',
+              'type': 'servicebusqueue@v1',
+              'authorization': {
+                'type': 'connectionString',
+                'connectionString': 'Endpoint=sb://[namespance].servicebus.windows.net/;SharedAccessKeyName=xxx;SharedAccessKey=[key];EntityPath=[name]'
+              }
+            }"
+
+        - name: create an Service Bus Topic destination with json payload
+          text: >
+            az iot central export destination create
+            --app-id {appid}
+            --dest-id {destintionid}
+            --content "{
+              'displayName': 'Service Bus Topic',
+              'type': 'servicebustopic@v1',
```

Comment:
Is there a reason why type isnt just an overall parameter (and enum) to help with typos? We can also do small checks for correct required values (ex: adx needs a cluster name) based on the type

Reply:
We can also build up content for the user - have them provide 'type', 'display-name' (that can be auto populated based on the type), and 'properties' as a json containing all other arguments that they need (ex: authorization). Then we use the type to determine which fields we need to check in 'properties' json.

Reply:
Customer might use the payload the get from CLI or API, split here would make any simpler for them, and also there is new adding destination from backend, I wouldn't duplicate validation logic here.

Reply:
I would still separate out the content as much as possible - we can just start with content as a dictionary (of --content if they gave anything) and then for each parameter they specify (ex: --type servicebus) add or update the value in the content dictionary. This way they can copy paste the payload or have us help build it up if they do not want to deal with jsons

Reply:
I think the overall point here is we think we can improve and optimize devEx for the command patterns that currently rely majorly on explicit json payloads. There can be [work done](https://github.com/Azure/azure-iot-cli-extension/pull/463#discussion_r760636472) to support both modes of interaction where users can pass in a json payload or provides overrides/explicit values through parameters & switches.

If there are timing constraints, this can be done in a future iteration because these commands are classified as `preview`. But for GA we definitely want to revisit these topics.

Reply:
If that's the rule, I wouldn't be against it, already tried to split it into different parameters, but found several issues with passing in different piece of json and merge inside, here is part of the code:
![image](https://user-images.githubusercontent.com/14871527/144358560-044b13af-0705-4580-b7e2-77aa950f0950.png)

Case 1: cmd>```az iot central export destination create --app-id 09475c28-5213-4ed2-ba00-aa7641ad4a2e --dest-id dest2 --type webhook@v1 --name "webhook" --url "https://bing.com" --header '{"x-custom-region":"abc" }'```

Shows error: unrecognized arguments: }'

Case 2: cmd>```az iot central export destination create --app-id 09475c28-5213-4ed2-ba00-aa7641ad4a2e --dest-id dest2 --type webhook@v1 --name "webhook" --url "https://bing.com" --header "{'x-custom-region':{'value': 'westus'}}"```

Runs successfully

Case 3: cmd>```az iot central export destination create --app-id 09475c28-5213-4ed2-ba00-aa7641ad4a2e --dest-id dest2 --type webhook@v1 --name "webhook" --url "https://bing.com" --header "{'x-custom-region':{'value': 'westus', 'secret':true}}"```

Shows following error:
![image](https://user-images.githubusercontent.com/14871527/144358876-335924fb-d639-4ef6-87ee-aeb56143353d.png)

Case 3: cmd> ```az iot central export destination create --app-id 09475c28-5213-4ed2-ba00-aa7641ad4a2e --dest-id dest2 --type webhook@v1 --name "webhook" --url "https://bing.com" --header "{'x-custom-region':{'value': 'westus', 'secret':'false'}}"```

Shows following error:
![image](https://user-images.githubusercontent.com/14871527/144359222-e68a3518-169b-43a2-ae01-866b2dcb2a52.png)


Any suggestions for this?  looks like 'secret': true will cause failure, quite new to python, any help is appreciated.

Reply:
@digimaun , @vilit1

Reply:
true should be enclosed in quotes.

Here's some more documentation for json strings in bash/shell/ps:

https://docs.microsoft.com/en-us/cli/azure/use-cli-effectively#generic-update-arguments
https://github.com/Azure/azure-iot-cli-extension/wiki/Tips

Reply:
Different shells have different input rules (where issues primarily come up with json input) which is what the tips wiki tries to illustrate.

You are able to append `--debug` and see how the CLI has parsed the args which is helpful in debugging
![image](https://user-images.githubusercontent.com/28658112/144492551-91fb5602-da0e-489e-a87c-9ff3177dc067.png)

Reply:
If you are using `cmd` try  `--header "{\"x-custom-region\":{\"value\": \"westus\", \"secret\":true}}"`

Reply:
The advantage of file based json input, is you can avoid the shell syntax issues.

Reply:
Thanks @digimaun , @vilit1, these point are very helpful!